### PR TITLE
Jonatan/mon 2012 be onboarding implementation

### PR DESCRIPTION
**Pull request by JKL98ISR**

### CODE REVIEW

File: backend/client/deepchecks_client/core/api.py [hunk 1]
```diff
@@ -15,13 +15,12 @@
 from copy import copy
 from datetime import datetime
 
+import deepchecks_client
```
Comment (matanper) [inline backend/client/deepchecks_client/core/api.py:18]: why?

File: backend/deepchecks_monitoring/api/v1/onboarding.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+from deepchecks_monitoring.schema_models.model_version import ModelVersion
+
+from .router import router
+
+
+class Step(int, enum.Enum):
+    """Sort order of ingestion errors output."""
+
+    REGISTERED = 0
+    MODEL = 1
```
Comment (matanper) [inline backend/deepchecks_monitoring/api/v1/onboarding.py:36]: Model and version are the same step

File: backend/deepchecks_monitoring/api/v1/onboarding.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    if model is None:
+        return {'step': Step.REGISTERED}
+    latest_version_query = (sa.select(ModelVersion)
+                            .where(ModelVersion.model_id == model.id)
+                            .order_by(ModelVersion.end_time.desc()).limit(1)
+                            .options(selectinload(ModelVersion.model)))
+    latest_version: ModelVersion = (await session.execute(latest_version_query)).scalars().first()
+    if not latest_version:
+        return {'step': Step.MODEL}
+    if latest_version.start_time == pdl.datetime(3000, 1, 1):
```
Comment (matanper) [inline backend/deepchecks_monitoring/api/v1/onboarding.py:84]: feels more robust than a fixed value
```suggestion
   # if start time is after end time, it means no data has been ingested yet
    if latest_version.start_time > latest_version.end_time:
```

**Review by matanper — COMMENTED** [review]
Looks okay, but the steps are a bit weird. I think the step should indicate what's the user currently missing, not the last thing he has.
also from the design model + version are single step

**Comment by JKL98ISR** [comment]
fixed it @matanper. @lironhamra can you check for a sec too?

**Review by matanper — APPROVED** [review]

**Review by matanper — APPROVED** [review]

**Pull request merged**